library/alloc/benches/vec.rs

```diff
@@ -732,3 +732,9 @@ fn bench_flat_map_collect(b: &mut Bencher) {
     let v = vec![777u32; 500000];
     b.iter(|| v.iter().flat_map(|color| color.rotate_left(8).to_be_bytes()).collect::<Vec<_>>());
 }
+
+#[bench]
+fn bench_retain_whole_100000(b: &mut Bencher) {
```

Comment:
Could you more bench(s) for case of many moves? Like retaining the second half of a Vec, or only even positions.